feat!: change generateTitle default to false

TylerBarnes opened the pull request:
Changed the default value of `threads.generateTitle` from `true` to `false`. Automatic title generation now requires explicit opt-in to avoid unexpected LLM API calls. I found it confusing when this would happen by default in playground. Sometimes I would be debugging the title llm call instead of the main one (I also saw others doing this)

Before:
```typescript
// Title generation happened automatically
const agent = new Agent({
  memory: new Memory({})
});
```

After:
```typescript
// Explicitly enable if you want title generation
const agent = new Agent({
  memory: new Memory({
    options: {
      threads: {
        generateTitle: true
      }
    }
  })
});
```

Also updated the playground UI to display thread IDs (last 5 chars) instead of "Chat from" when titles aren't generated, making it easier to identify threads.

<img width="291" height="294" alt="Screenshot 2025-10-20 at 4 53 32 PM" src="https://github.com/user-attachments/assets/c4de5e07-18c5-49fd-abf5-54253f4c40af" />

Keeping this a draft for now until after this weeks release. I have multiple breaking change PRs for memory that I'm getting queued up.

TylerBarnes committed:
feat!: change generateTitle default from true to false

BREAKING CHANGE: Thread title auto-generation is now opt-in

Changed the default value of threads.generateTitle from true to false
to make the framework more cost-conscious and predictable. Automatic
title generation now requires explicit opt-in to avoid unexpected
LLM API calls.

Changes:
- Updated default in packages/core/src/memory/memory.ts
- Updated English and Japanese documentation
- Added migration guide
- Added explanatory comments to all examples showing generateTitle: true

To maintain previous behavior, explicitly set:
threads: { generateTitle: true }

Closes #XXX

TylerBarnes committed:
display ID instead of "Chat from"

TylerBarnes committed:
simplify

TylerBarnes committed:
changeset

TylerBarnes committed:
Merge branch 'main' into break/no-generate-title-playground

TylerBarnes committed:
docs: merge generateTitle migration into v1.0 guide

TylerBarnes committed:
Merge remote-tracking branch 'origin/main' into break/no-generate-title-playground

TylerBarnes committed:
feat!: move generateTitle to top-level memory option

BREAKING CHANGE: Moved generateTitle from threads.generateTitle to top-level memory option. Using threads.generateTitle will now throw an error.

- Move generateTitle to top-level MemoryConfig
- Deprecate threads.generateTitle with never type
- Add runtime error when threads.generateTitle is used
- Update all code references to use new location
- Update tests to use new API
- Update playground UI to read from new location
- Update migration guide with new API structure

TylerBarnes committed:
Merge branch 'main' into break/no-generate-title-playground

TylerBarnes committed:
prettier

TylerBarnes committed:
Merge branch 'main' into break/no-generate-title-playground

TylerBarnes committed:
fix: update @mastra/ai-sdk peer dependency range to support v1.0.0-alpha

TylerBarnes committed:
fix: update all @mastra/core peer dependency ranges to support v1.0.0-alpha

TylerBarnes committed:
chore: update pnpm-lock.yaml after peer dependency changes

TylerBarnes committed:
fix: update processors.test.ts to use top-level generateTitle

TylerBarnes committed:
fix: update all memory integration tests to use top-level generateTitle

TylerBarnes committed:
fix: correct syntax errors from sed script in memory integration tests

TylerBarnes committed:
fix: remove extra closing brace in working-memory.test.ts

TylerBarnes committed:
fix: remove remaining extra closing braces in working-memory.test.ts

TylerBarnes committed:
fix: update MCP docs tests to handle missing examples and empty changelog sections

TylerBarnes committed:
fix: add type assertion for inputOptions.plugins to resolve lint error

TylerBarnes commented on packages/cli/src/commands/dev/DevBundler.ts:
```diff
@@ -107,9 +107,8 @@ export class DevBundler extends Bundler {
-
-          ...inputOptions.plugins,
+          // inputOptions.plugins is guaranteed to be Plugin[] by getWatcherInputOptions
+          ...((inputOptions.plugins as Plugin[]) || []),
```
@wardpeet local lint always fails on this too so claude ended up changing it

TylerBarnes committed:
fix: correct syntax error in working-memory.test.ts

Fixed missing closing brace and incorrect nesting of generateTitle option.

https://github.com/mastra-ai/mastra/pull/9045#discussion_r1872059326

TylerBarnes committed:
docs: move generateTitle to top-level in threads-and-resources.mdx

Updated API usage from threads.generateTitle to top-level generateTitle.

https://github.com/mastra-ai/mastra/pull/9045#discussion_r1872059327

TylerBarnes committed:
docs: move generateTitle to top-level in working-memory-basic.md

Updated API usage from threads.generateTitle to top-level generateTitle.

https://github.com/mastra-ai/mastra/pull/9045#discussion_r1872059328

TylerBarnes committed:
docs: move generateTitle to top-level in memory-processors.md

Updated API usage from threads.generateTitle to top-level generateTitle.

https://github.com/mastra-ai/mastra/pull/9045#discussion_r1872059329

TylerBarnes committed:
docs: move generateTitle to top-level in memory-with-libsql.mdx

Updated API usage from threads.generateTitle to top-level generateTitle.

https://github.com/mastra-ai/mastra/pull/9045#discussion_r1872059330

TylerBarnes committed:
docs: move generateTitle to top-level in memory-with-pg.mdx

Updated API usage from threads.generateTitle to top-level generateTitle.

https://github.com/mastra-ai/mastra/pull/9045#discussion_r1872059331

TylerBarnes committed:
docs: move generateTitle to top-level in memory-with-upstash.mdx

Updated API usage from threads.generateTitle to top-level generateTitle.

https://github.com/mastra-ai/mastra/pull/9045#discussion_r1872059332

TylerBarnes commented:
Reese says: @greptileai Thanks for the thorough review! I've addressed all the issues you identified:

1. ✅ Fixed syntax error in `working-memory.test.ts` - corrected the missing closing brace and nesting
2. ✅ Updated all 6 documentation files to use the new top-level `generateTitle` API instead of the deprecated `threads.generateTitle` path

All changes have been committed separately with links to the specific review comments. The code should now be safe to merge.

Reese

TylerBarnes committed:
chore: trigger CI rebuild

TylerBarnes committed:
Merge branch 'main' into break/no-generate-title-playground

TylerBarnes committed:
fix: move generateTitle from threads to top-level in memoryDefaultOptions

TylerBarnes committed:
fix: update integration-tests-v5 to use top-level generateTitle

LekoArts reviewed (commented):
You can drop all changes to the `ja` docs 😅

TylerBarnes commented:
> You can drop all changes to the `ja` docs 😅

@LekoArts I know they're generated, claude updated those and I didn't think it really mattered since they'll be regenerated anyway

TylerBarnes committed:
Merge main into break/no-generate-title-playground

TylerBarnes committed:
Merge remote-tracking branch 'origin/main' into break/no-generate-title-playground

TylerBarnes committed:
Merge main into break/no-generate-title-playground

TylerBarnes committed:
Fix: Remove merge conflict marker from migration guide

TylerBarnes committed:
fix(e2e): enable generateTitle for weather agent E2E tests

The E2E tests expect thread titles to be generated, so we need to
explicitly enable generateTitle for the weather agent used in tests.

TylerBarnes committed:
fix(e2e): update stream test to not rely on thread titles

The test was incorrectly assuming that thread titles would be the full
response text. With generateTitle now defaulting to false, threads won't
have titles by default. Updated the test to just check that a thread link
exists in the list, rather than checking for a specific title.

TylerBarnes committed:
fix(tests): update agent tests to use top-level generateTitle
